### PR TITLE
Added EC2 launch type for China region

### PR DESCRIPTION
**Pull request by azeliashchonak**

### CODE REVIEW

File: ecs-cluster-ec2.tf [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  provisioner "local-exec" {
+    command = "sleep 60"
+  }
+
+  count = "${var.launch_type == "FARGATE" ? 0 : 1}"
+}
+
+resource "aws_launch_configuration" "launch-configuration_ec2" {
+  name                 = "${var.service}-launch-configuration"
+  image_id             = "ami-0da0590b87d9022f2"
```
[the remark this answers is not in the document]
Comment (azeliashchonak) [reply ecs-cluster-ec2.tf]: fixed

File: ecs-cluster-ec2.tf [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  instance_type        = "${var.instance_type}"
+  iam_instance_profile = "${aws_iam_instance_profile.ecs-instance-profile_ec2.id}"
+  key_name             = "${var.key-pair-name}"
+
+  root_block_device {
+    volume_type           = "standard"
+    volume_size           = 100
```
[the remark this answers is not in the document]
Comment (azeliashchonak) [reply ecs-cluster-ec2.tf]: fixed

File: ecs-cluster.tf [hunk 3]
```diff
@@ -37,7 +37,7 @@ resource "aws_ecs_service" "this" {
   name                               = "${var.service}-${var.environment}"
   cluster                            = "${local.ecs_cluster_id}"
   task_definition                    = "${aws_ecs_task_definition.this.arn}"
-  launch_type                        = "FARGATE"
+  launch_type                        = "${var.launch_type == "FARGATE" ? "FARGATE" : "EC2"}"
```
[the remark this answers is not in the document]
Comment (azeliashchonak) [reply ecs-cluster.tf]: fixed

File: output.tf [hunk 4]
```diff
@@ -1,10 +1,10 @@
 output "ecs_service_iam_role_arn" {
-  value       = "${aws_iam_role.ecs-service.arn}"
+  value       = "${var.launch_type == "FARGATE" ? "${element(concat(aws_iam_role.ecs-service.*.arn, list("")), 0)}":"${element(concat(aws_iam_role.ecs-service-ec2.*.arn, list("")), 0)}"}"
```
[the remark this answers is not in the document]
Comment (azeliashchonak) [reply output.tf]: fixed

File: output.tf [hunk 5]
```diff
@@ -1,10 +1,10 @@
...
   description = "ARN fo created ECS service"
 }
 
 output "ecs_service_iam_role_name" {
-  value       = "${aws_iam_role.ecs-service.name}"
+  value       = "${var.launch_type == "FARGATE" ? "${element(concat(aws_iam_role.ecs-service.*.name, list("")), 0)}":"${element(concat(aws_iam_role.ecs-service-ec2.*.name, list("")), 0)}"}"
```
[the remark this answers is not in the document]
Comment (azeliashchonak) [reply output.tf]: fixed

File: variables.tf [hunk 6]
```diff
@@ -101,4 +101,23 @@ variable "subnets" {
...
+}
+
+variable "instance_type" {
+  description = "ARN of target group"
+  type        = "string"
+  default     = "t2.small"
+}
+
+variable "launch_type" {
+  description = "Launch type for ECS"
```
[the remark this answers is not in the document]
Comment (azeliashchonak) [reply variables.tf]: fixed

File: output.tf [hunk 7]
```diff
@@ -67,3 +67,8 @@ output "security_group_vpc_id" {
   description = "The VPC ID."
   value       = "${module.security-group.this_security_group_vpc_id}"
 }
+
+output "vpc_id" {
```
[the remark this answers is not in the document]
Comment (azeliashchonak) [reply output.tf]: fixed

File: autoscaling.tf [hunk 8]
```diff
@@ -89,3 +127,41 @@ resource "aws_appautoscaling_target" "ecs_target" {
...
+  count                  = "${var.launch_type == "FARGATE" ? 0 : 1}"
+}
+
+resource "aws_autoscaling_group" "autoscaling-group" {
+  name             = "${var.service}-autoscaling-group"
+  max_size         = "3"
+  min_size         = "1"
+  desired_capacity = "1"
+
+  availability_zones   = ["cn-north-1a", "cn-north-1b"]
```
Comment (ToROxI) [inline autoscaling.tf]: Please avoid hardcode

File: define-policies.tf [hunk 9]
```diff
@@ -67,17 +67,67 @@ resource "aws_iam_role" "ecs-service" {
...
+  name = "ecs-service-ec2-${var.project}-${var.service}-${var.environment}"
+
+  assume_role_policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Action": "sts:AssumeRole",
+      "Principal": {
+        "Service": "ec2.amazonaws.com.cn"
```
Comment (ToROxI) [inline define-policies.tf]: Please avoid hardcode

File: define-policies.tf [hunk 10]
```diff
@@ -67,17 +67,67 @@ resource "aws_iam_role" "ecs-service" {
...
+  ]
+}
+EOF
+
+  count = "${var.launch_type == "FARGATE" ? 0 : 1}"
+  tags  = "${merge(local.default_tags, var.tags)}"
+}
+
+resource "aws_iam_role_policy_attachment" "this_ec2" {
+  policy_arn = "arn:aws-cn:iam::aws:policy/service-role/AmazonEC2ContainerServiceforEC2Role"
```
Comment (ToROxI) [inline define-policies.tf]: Please avoid hardcode

File: define-policies.tf [hunk 11]
```diff
@@ -67,17 +67,67 @@ resource "aws_iam_role" "ecs-service" {
...
+  role       = "${aws_iam_role.ecs-service-ec2.name}"
+  count      = "${var.launch_type == "FARGATE" ? 0 : 1}"
+}
+
+resource "aws_iam_role_policy_attachment" "this_default_ecs_ec2" {
+  policy_arn = "arn:aws-cn:iam::aws:policy/service-role/AmazonEC2ContainerServiceRole"
```
Comment (ToROxI) [inline define-policies.tf]: Please avoid hardcode

File: autoscaling.tf (hunk 8 above)
Comment (azeliashchonak) [reply autoscaling.tf]: fixed

File: define-policies.tf (hunk 9 above)
Comment (azeliashchonak) [reply define-policies.tf]: fixed

File: define-policies.tf (hunk 11 above)
Comment (azeliashchonak) [reply define-policies.tf]: fixed

File: define-policies.tf (hunk 10 above)
Comment (azeliashchonak) [reply define-policies.tf]: fixed

**Pull request merged**